### ISSUE TITLE
Assert in SearchData::addAll

### ISSUE THREAD

**Issue by xhargh** [variar/klogg#199]
How to reproduce, latest master as of today (2020-07-06), using Qt 5.15.0 on kubuntu 20.04:

 * open a log file that is updated in the background. I've tested with `/tmp/foo` which I update with this:
```
while(true); do fortune | cowsay -f $(ls /usr/share/cowsay/cows | shuf -n1) >> /tmp/foo ; sleep 1; done
```
 * search for something that occurs often in the file, I have used `oo`
 * auto-refresh should be enabled
 * manually refresh repeatedly until the manual refresh collides with auto-refresh and there is new matching data in the log file.

I get the assert:

```
klogg: .../src/logdata/src/logfiltereddataworker.cpp:148: SearchData::addAll(LineLength, const SearchResultArray&, LinesCount)::<lambda(SearchResultArray&)>: Assertion `insertIt == end( oldMatches ) || !( *insertIt < matches.back() )' failed.
```

I've not yet figured out what's wrong...

**Comment by variar** [comment]
Looks like data race. I hardly ever use auto-refresh, so this area is not tested much. I'll try to find the cause.

**Comment by variar** [comment]
@xhargh unfortunately I could not reproduce this crash. And thread sanitizer is not very helpful because of false positives about TBB and Qt. However, I've found one data race that could lead to the crash: index data could be cleared and reloaded while search thread was reading next chunk of lines. I'm not sure if this race is entirely fixed but it should appear much less often. Could you try builds 20.9.0.670+?

**Comment by xhargh** [comment]
@variar, I pulled the latest commit from master and tested, I can still reproduce the issue.

![callstack 9b344976e8854c601d8b1078b9ab61d472982245](https://user-images.githubusercontent.com/8505258/93777117-a2e28d80-fc24-11ea-8250-5d300e4c3063.png)

**Comment by variar** [comment]
Well, looks like threading times in debugger are more prone to the race.

**Comment by variar** [comment]
My bad. Assert is disabled in release builds. This is actually easy to reproduce.

**Comment by variar** [comment]
This assert turned out to be unrelated with fixes in clearing index data during search. The problem was in starting position for next search update. For some unknown reason this start position was taken from current search operation progress shown in UI. So in case of manual research -- from arbitrary point inside a file.

Should be fixed in 20.9.0.672+

@xhargh thanks a lot for providing steps to reproduce this issue. This was a major bug. So next release should be much better!

**Comment by xhargh** [comment]
I did a quick test and I couldn't reproduce the issue. Thanks a lot!
